sound/soc/sof/intel/hda-dsp.c

```diff
@@ -405,6 +405,26 @@ static int hda_dsp_update_d0i3c_register(struct snd_sof_dev *sdev, u8 value)
 	return 0;
 }
 
+static bool hda_dsp_d0i3_streaming(struct snd_sof_dev *sdev)
+{
+	struct snd_pcm_substream *substream;
+	struct snd_sof_pcm *spcm;
+	int dir;
+
+	list_for_each_entry(spcm, &sdev->pcm_list, list) {
+		for_each_pcm_streams(dir) {
```

Comment:
since you only care about playback, why check PLAYBACK and CAPTURE? Just drop this for_each_pcm_streams and use SNDRV_PCM_STREAM_PLAYBACK directly in place of 'dir'.
Btw, is this going to be applicable for compressed playback as well?

Reply:
sure. thanks!   It is not relate to compressed or non-compressed stream.